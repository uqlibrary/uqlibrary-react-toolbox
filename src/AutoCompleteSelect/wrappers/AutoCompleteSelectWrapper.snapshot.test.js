jest.dontMock('./AutoCompleteSelect');

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import AutoCompleteSelectWrapper from './AutoCompleteSelect';
import filterProps from '../../helpers/_filterProps';

function setup(props) {
    const consolidatedProps = filterProps(props, AutoCompleteSelectWrapper.propTypes);
    consolidatedProps.onTouchTap = props.input.onBlur;
    consolidatedProps.error = !!consolidatedProps.errorText;

    return shallow(<AutoCompleteSelectWrapper {...props} />);
}

describe('AutoCompleteSelectWrapper snapshots tests', () => {
    it('renders autocomplete field', () => {
        const props =
            {
                name: 'owner',
                label: 'Owner',
                disabled: false,
                filterItems: jest.fn(),
                labelField: 'name',
                value: {},
                errorText: '',
                meta: {
                    touched: null
                },
                input: {
                    onBlur: null
                }
            };

        const wrapper = setup(props);
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });
});
