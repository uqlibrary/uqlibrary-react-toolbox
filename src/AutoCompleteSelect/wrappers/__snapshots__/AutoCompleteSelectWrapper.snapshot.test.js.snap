// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoCompleteSelectWrapper snapshots tests renders autocomplete field 1`] = `
<div
  className="flex column"
>
  <div
    style={
      Object {
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <AsyncAutoCompleteSelect
      debounceDuration={300}
      disabled={false}
      emptySearchText="Start typing to filter data..."
      error={false}
      filterItems={[Function]}
      label="Owner"
      labelField="name"
      minLength={3}
      name="owner"
      noResultsText="No results found with those search details"
      onBlur={null}
      onTouchTap={null}
      value={Object {}}
    />
  </div>
</div>
`;
